src/NUnitFramework/framework/Internal/Execution/CompositeWorkItem.cs

```diff
@@ -232,12 +232,29 @@ private void RunChildren()
         private void CreateChildWorkItems()
         {
             _children = new List<WorkItem>();
-
+            
             foreach (ITest test in _suite.Tests)
                 if (_childFilter.Pass(test))
                     _children.Add(WorkItem.CreateWorkItem(test, _childFilter));
+            Sort();
         }
 
+        /// <summary>
+        /// Sorts tests under this suite.
+        /// </summary>
+        private void Sort()
+        {
+                this._children.Sort(delegate (WorkItem x, WorkItem y)
```

Comment:
SortChildren? Coding standard: don't use this. The leading underscore is sufficient to make it clear.